src/main/java/org/openrewrite/java/testing/mockito/PowerMockitoMockStaticToMockito.java

```diff
@@ -327,154 +297,240 @@ private static List<Expression> getMockedTypesFromPrepareForTestAnnotation(List<
+            }
 
-                if (mockedTypesIdentifiers != null) {
-                    for (J.Identifier mockedTypesField : mockedTypesIdentifiers) {
-                        // Only add close method invocation if not already exists
-                        J.Block methodBody = m.getBody();
-                        if (methodBody == null || isStaticMockAlreadyOpened(mockedTypesField, methodBody)) {
-                            continue;
-                        }
+            this.setUpMethodAnnotation = "@" + setUpMethodAnnotationName;
+            this.tearDownMethodAnnotation = "@" + tearDownMethodAnnotationName;
 
-                        String className = ((JavaType.Class) ((JavaType.Parameterized) mockedTypesField.getType()).getTypeParameters()
-                          .get(0)).getClassName();
-                        m = m.withBody(methodBody.withTemplate(
-                          JavaTemplate.builder(() -> getCursor().getParentTreeCursor(),
-                              "mocked#{any(org.mockito.MockedStatic)} = mockStatic(#{}.class);")
-                            .javaParser(() -> JavaParser.fromJavaVersion()
-                              .classpathFromResources(ctx, "mockito-core-3.*")
-                              .build())
-                            .build(),
-                          methodBody.getCoordinates().lastStatement(),
-                          mockedTypesField,
-                          className
-                        ));
-                    }
-                }
-                // The mocked field are needed for other visitors
-                setMockedTypesFields(mockedTypesIdentifiers);
-                return m;
-            }
-            return m;
+            this.setUpMethodAnnotationSignature = "@" + annotationPackage + "." + setUpMethodAnnotationName;
+            this.tearDownMethodAnnotationSignature = "@" + annotationPackage + "." + tearDownMethodAnnotationName;
+
+            this.setUpImportToAdd = annotationPackage + "." + setUpMethodAnnotationName;
+            this.tearDownImportToAdd = annotationPackage + "." + tearDownMethodAnnotationName;
         }
 
-        private void setMockedTypesFields(List<J.Identifier> mockedTypesFields) {
-            this.mockedTypesFields = mockedTypesFields;
+        private Map<J.Identifier, Expression> getMockedTypesFields() {
+            if (mockedTypesFields == null) {
+                mockedTypesFields = getCursor().getNearestMessage(MOCKED_TYPES_FIELDS, new LinkedHashMap<>());
+            }
+            return mockedTypesFields;
         }
 
-        private static boolean isStaticMockAlreadyClosed(J.Identifier staticMock, J.Block methodBody) {
-            return methodBody.getStatements().stream().filter(statement -> statement instanceof J.MethodInvocation)
+        private void findMockStaticInvocations(J.ClassDeclaration classDecl) {
+            // Interfaces have null bodies
```

Comment:
Not always the case, there can be `default` method implementations in interfaces